### PR TITLE
Switched to .NET 9.0 System.Threading.Lock

### PR DESCRIPTION
**Pull request by MarkCiliaVincenti**

### CODE REVIEW

**Comment by dj-nitehawk** [comment]
thanks for the PR but...
you're bringing in a third party dependency `Backport.System.Threading.Lock` to add support for pre .net 9 sdks yes?
we definitely don't wanna do that for something that doesn't offer much of a benefit.
if you want, you can just do conditional compilation blocks to support the new `Lock` type on .net 9 onwards.

**Comment by MarkCiliaVincenti** [comment]
> thanks for the PR but... you're bringing in a third party dependency `Backport.System.Threading.Lock` to add support for pre .net 9 sdks yes? we definitely don't wanna do that for something that doesn't offer much of a benefit. if you want, you can just do conditional compilation blocks to support the new `Lock` type on .net 9 onwards.

Yes, that's what the micro dependency basically does, but also allows you to use the full functionality of the Lock class.

File: Src/Messaging/Messaging.Remote/Server/Events/EventHub.cs [hunk 1]
```diff
@@ -44,7 +44,14 @@ sealed class EventHub<TEvent, TStorageRecord, TStorageProvider> : EventHubBase,
     static readonly Type _tEvent = typeof(TEvent);
     static bool _isRoundRobinMode;
     static TStorageProvider? _storage;
-    static readonly Lock _lock = new();
+
+    readonly
```
Comment (MarkCiliaVincenti) [inline Src/Messaging/Messaging.Remote/Server/Events/EventHub.cs:48]: This is no longer static

Comment (dj-nitehawk) [reply Src/Messaging/Messaging.Remote/Server/Events/EventHub.cs:48]: oops...
good catch!
fixed.

**Comment by MarkCiliaVincenti** [comment]
@dj-nitehawk a new version of Backport.System.Threading.Lock has come out that acts as a source generator and basically dropping the DLL as a dependency. If this new development makes you reconsider, I will amend accordingly.

**Comment by dj-nitehawk** [comment]
will check what the source generator creates and get back to you on that.

**Comment by MarkCiliaVincenti** [comment]
Handled via #824

**Pull request merged**